test/spec/controllers/notificationCtrlSpec.js

```diff
@@ -1,33 +1,50 @@
 'use strict';
 
 describe('Notification Controller', function() {
-  var user, scope, rootScope, ctrl;
+  var user, scope, rootScope, fakeBackend, achievement, ctrl;
 
   beforeEach(function() {
     user = specHelper.newUser();
     user._id = "unique-user-id";
 
+    var userSync = sinon.stub().returns({
+      then: function then (f) { f(); }
```

Comment:
@crookedneighbor I tried that, but I need the function that gets passed into the `then` to actually get called, otherwise the `wonChallenge` test fails.